ipfshttpclient/http.py

```diff
@@ -30,4 +33,7 @@
 	except ImportError:  # pragma: no cover
 		from . import http_httpx as _backend
 
-ClientSync = _backend.ClientSync
+
+# noinspection PyPep8Naming
+def ClientSync(*args, **kwargs) -> ClientSyncBase[ty.Any]:  # type: ignore[no-untyped-def]
+	return _backend.ClientSync(*args, **kwargs)
```

Comment:
Given that `ClientSync` is a private implementation detail of the library, and only called from `class ClientBase` - we arn't constrained to support backward compatibility on it like we would a class such as `Client`.

The gain here is a warning-free `Sphinx` documentation generation, which enables warnings to be configured as errors, thus catching problems during the build and publish that would otherwise be caught after docs have been made public.

As far as the specific approach I took (`def ClientSync(*args ..etc)`) I agree it is.... not pretty. I explored other approaches before choosing this one; they either broke Sphinx, or they required a larger code change to the rest of the library.

The current approach minimizes the amount of code change. The larger changes involved renaming `ClientSync` to look like a factory/builder method (small/moderate change) or to introduce another abstract base class (large change, seemed like a solution worse than the original problem).

Do we need to maintain two http client implementations going forward? Is that an artifact of supporting Python versions that are no longer supported (such as Python 3.5)?

Reply:
> This is a bad idea, since people can do:
> 
> ```python
> class MyClient(ipfshttpclient.Client):
>     def some_extra_method(self):
>         …
> ```
> 
> and this change would break that IMHO for no gain at all.

`ipfshttpclient.Client` does not inherit from `ClientSync`:

`class Client(files.Base, miscellaneous.Base):`

`files.Base` inherits from `class Base(base.ClientBase):` which creates an instance of `ClientSync` (or in this PR, calls it as a factory method). I.e., `ClientBase` is getting a `ClientSync` via composition instead of inheritance.

Reply:
> The gain here is a warning-free Sphinx documentation generation, which enables warnings to be configured as errors, thus catching problems during the build and publish that would otherwise be caught after docs have been made public.

What kind of warning is it anyways? Assuming its a dumb one that doesn't help anybody: Can't we just ignore the warning instead?

> Do we need to maintain two http client implementations going forward? Is that an artifact of supporting Python versions that are no longer supported (such as Python 3.5)?

Well, HTTPx never supported Python 3.5 so that's why we have this solution in the first place. I'm a bit hasitant to go all-in on HTTPx through until it gets out of beta – its APIs are still theoretically subject to change after all.
This change was actually done in preparation of adding `async`-support to the library – something that requests doesn't support and may never will.

Reply:
Thanks for the explanation on HTTPx, I understand that better now.